servicetalk-opentracing-http/src/main/java/io/servicetalk/opentracing/http/TracingHttpServiceFilter.java

```diff
@@ -40,6 +43,12 @@
 
 /**
  * A {@link StreamingHttpService} that supports open tracing.
+ * <p>
+ * Append this filter before others that are expected to see {@link Scope} for this request/response. Filters appended
```

Comment:
discussed with @NiteshKant offline on possibility of splitting this filter into two:
- TracerExtractor
- TracerCloser

however the complexity of implementation would be high to hand-off ownership between filters. Also I will come up with a PR to change `when*` operators to default to `before*` (instead of `after*`) to reduce the risk of filters inadvertently applying an "after" operator which would mean their filter response processing may run without trace info.

Reply:
Yes agreed! 

It seems logical to define the scope of an operation within the context of a filter. `After*` operators seem to extend that scope after everything else outside that scope. Drawing an analogy to an imperative programming model:

```java
public Result someMethod(Caller caller) {

    State state;
    try {
        state = new State();
        Result result = doWork();
        return result;
    } finally {
         // Code here is analogous to beforeFinally async operators.
         state.cleanup(); 
        // In order to demonstrate cleanup from the code that is calling this method. 
        // Assume caller is passed to this method.
        caller.cleanup();
    }
// Code below this is analogous to async afterFinally operators.
}
```